dspace-api/src/main/java/org/dspace/app/requestitem/RequestItemMetadataStrategy.java

```diff
@@ -37,52 +40,64 @@ public RequestItemMetadataStrategy() {
     }
 
     @Override
-    public RequestItemAuthor getRequestItemAuthor(Context context, Item item)
+    public List<RequestItemAuthor> getRequestItemAuthor(Context context, Item item)
         throws SQLException {
-        RequestItemAuthor author = null;
+        List<RequestItemAuthor> authors;
         if (emailMetadata != null) {
             List<MetadataValue> vals = itemService.getMetadataByMetadataString(item, emailMetadata);
             if (vals.size() > 0) {
-                String email = vals.iterator().next().getValue();
-                String fullname = null;
-                if (fullNameMetadata != null) {
-                    List<MetadataValue> nameVals = itemService.getMetadataByMetadataString(item, fullNameMetadata);
-                    if (nameVals.size() > 0) {
-                        fullname = nameVals.iterator().next().getValue();
+                authors = new ArrayList<>(vals.size());
+                for (MetadataValue datum : vals) {
+                    String email = datum.getValue();
+                    String fullname = null;
+                    if (fullNameMetadata != null) {
+                        List<MetadataValue> nameVals = itemService.getMetadataByMetadataString(item, fullNameMetadata);
+                        if (!nameVals.isEmpty()) {
+                            fullname = nameVals.get(0).getValue();
```

Comment:
this is the usual problem due to lack of support in DSpace for hierarchical metadata. We are looking to two different metadata for piece of information that need to be read together. I suggest one of this two approaches
- (my preference) check if the vals and nameVals have the same size, if so it is ok to read them in pair. If not we should just ignore the fullname and take only the emails
- alternatively we should hardcode the rule that this strategy will only return the first value

Reply:
Your preferred resolution makes more sense than what I had.  Done.